pulsar-client-cpp/include/pulsar/ReaderConfiguration.h

```diff
@@ -86,6 +86,9 @@ class ReaderConfiguration {
     void setSubscriptionRolePrefix(const std::string& subscriptionRolePrefix);
     const std::string& getSubscriptionRolePrefix() const;
 
+    void setReadCompacted(bool compacted);
```

Comment:
Why isn't this returning ReaderConfiguration&? I see some others are not returning it, but I would guess it's to not break BC. setReaderListener does return it.